CommonTools/RecoAlgos/src/ClusterStorer.cc

```diff
@@ -166,7 +166,7 @@ namespace helper {
       SiStripMatchedRecHit2D &mhit = static_cast<SiStripMatchedRecHit2D &>(genericHit);
       cluRef = (SiStripDetId(detid_).stereo() ? &mhit.stereoClusterRef() : &mhit.monoClusterRef());
     } else if (typeid(ProjectedSiStripRecHit2D) == hit_type) {
-      cluRef = &static_cast<ProjectedSiStripRecHit2D &>(genericHit).originalHit().omniCluster();
+      cluRef = &static_cast<ProjectedSiStripRecHit2D &>(genericHit).omniCluster();
```

Comment:
Uhm, what was the original purpose of the now removed `originalHit()`?
Maybe @mmusich has some idea...

Reply:
I am not sure about this one.
I see that the `ProjectedSiStripRecHit2D::originalHit()` method was added by @VinInn in https://github.com/cms-sw/cmssw/pull/2789 (8 years ago) with  this comment:

https://github.com/cms-sw/cmssw/blob/4e89b5b212dc118b0b17abffd052144eea7ebe04/DataFormats/TrackerRecHit2D/interface/ProjectedSiStripRecHit2D.h#L55-L56

perhaps Vincenzo has more insight here.

Reply:
Clearly the change is modifying behaviour. Do not really understand why was made in first place....

Reply:
So looking at the code, the present code is effectively a 'no-op'. The call to `originalHit()` creates a temporary which has its own copy of the `omniCluster`. So on line 174

https://github.com/cms-sw/cmssw/blob/1ae5adae409a1971bf448456dc8d581e608a47d3/CommonTools/RecoAlgos/src/ClusterStorer.cc#L174

the object owned by the temporary is modified and then the temporary goes away at the end of the routine and the change object is also discarded.

So to keep the old implementation one would just remove line 169.

Reply:
> So to keep the old implementation one would just remove line 169.

and then hit the assert in L172?

Reply:
> and then hit the assert in L172?

That's true. Than change like 169 to `return;`

Reply:
I'm trying to recall if ProjectedSiStripRecHit2D is actually used

Reply:
> I'm trying to recall if ProjectedSiStripRecHit2D is actually used

I found it in somewhat random local igprof log. So, not a dead end. That was not including this code in this PR in the path though.

Reply:
The current behavior is undefined because `originalHit()` returns a temporary, that holds a copy of the `genericHit.omniClusterRef()`
https://github.com/cms-sw/cmssw/blob/4e89b5b212dc118b0b17abffd052144eea7ebe04/DataFormats/TrackerRecHit2D/interface/SiStripRecHit2D.h#L16
https://github.com/cms-sw/cmssw/blob/4e89b5b212dc118b0b17abffd052144eea7ebe04/DataFormats/TrackerRecHit2D/interface/TrackerSingleRecHit.h#L23

I'd say the question is whether the behavior proposed in this PR (that is to rekey the `genericHit.omniClusterRef()`) makes sense.

Reply:
> I am not sure about this one. I see that the `ProjectedSiStripRecHit2D::originalHit()` method was added by @VinInn in #2789 (8 years ago) with this comment:
> 
> https://github.com/cms-sw/cmssw/blob/4e89b5b212dc118b0b17abffd052144eea7ebe04/DataFormats/TrackerRecHit2D/interface/ProjectedSiStripRecHit2D.h#L55-L56
> 
> perhaps Vincenzo has more insight here.

I navigated to the original change https://github.com/cms-sw/cmssw/commit/39d8c28aebba932ed832e08a716cb28695dd33be 
at that point the `originalHit` was present in the hit

In #2789 the `originalHit_` was removed from `ProjectedSiStripRecHit2D` data members 
https://github.com/cms-sw/cmssw/commit/f32479268c995140cebe1d72a5fe102a19824b96#diff-69366d848e58d8e26caac67c34f291dd469ca07b11374db0e4a44aac928b7ae0L26-R54

The clusterRef that was in the copied hit `originalHit_` still remains in the base data member (`TrackerSingleRecHit::cluster_`).  The current variant that creates a temporary  and then calls `SiStripRecHit2D::omniCluster` would get to the same value (the call via `.originalHit` simply passes the cluster to the temporary's base `TrackerSingleRecHit::cluster_`).

So, I think that the proposed solution is correct.